applications/GeoMechanicsApplication/tests/cpp_tests/test_time_incrementor.cpp

```diff
@@ -386,4 +397,17 @@ KRATOS_TEST_CASE_IN_SUITE(ReduceUpscaledIncrementToAvoidExceedingEndTime, Kratos
     KRATOS_EXPECT_DOUBLE_EQ(0.5 * settings.StartIncrement, time_incrementor.GetIncrement());
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ReduceUpscaledIncrementToAvoidExceedingMaxDeltaTimeFactor, KratosGeoMechanicsFastSuite)
+{
+    AdaptiveTimeIncrementorSettings settings;
+    settings.MaxDeltaTimeFactor = 1.5; // lower than the increase factor, this should lead to truncation
+    auto time_incrementor = MakeAdaptiveTimeIncrementor(settings);
+    auto previous_state   = TimeStepEndState{};
+    previous_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    previous_state.num_of_iterations = settings.MinNumOfIterations - 1; // this should normally increase the increment
+
+    time_incrementor.PostTimeStepExecution(previous_state);
+    KRATOS_EXPECT_DOUBLE_EQ(1.5 * settings.StartIncrement, time_incrementor.GetIncrement());
```

Comment:
Perhaps the following is slightly clearer?
```suggestion
    KRATOS_EXPECT_DOUBLE_EQ(settings.MaxDeltaTimeFactor * settings.StartIncrement, time_incrementor.GetIncrement());
```

Reply:
done